bmaptools/BmapCopy.py

```diff
@@ -404,6 +427,16 @@ def _update_progress(self, blocks_written):
         self._progress_file.write(progress)
         self._progress_file.flush()
 
+        # Update psplash progress when configured. This is using a best effort
+        # strategy to not affect the writing process when psplash breaks, is
+        # not available early enough or screen is not available.
+        if self._psplash_pipe and self.mapped_cnt:
+            try:
+                with os.fdopen(os.open(self._psplash_pipe, os.O_WRONLY | os.O_NONBLOCK), 'w') as p:
+                    p.write("PROGRESS %d\n" % percent)
+            except:
+                pass
```

Comment:
Err, not liking this. If the pipe can't be open or written to, we'll end up with 2 imperfect things:
1. We'll return here again and again and will fail again and again. Set self._psplash_pipe to None to prevent this.
2. We'll just silently ignore the error, giving the user no indication that --psplash does not work. Print a warning here instead. And the above step will make sure the warning is printed only once.

Reply:
The fact that we failed once doesn't mean it will fail the next time. For example, that is the case where at boot you have psplash and a bmaptool process but psplash becomes ready a bit later because it needs to wait for the framebuffer to get initialized.

There can be cases where the framebuffer never gets up. For example, no screen is attached at a specific boot. You wouldn't want to break a flashing boot functionality for a decoration feature. So, the use case here is that you can flash a disk at boot with psplash report when that is available and if that becomes available. The alternative would be to have a much stricter syncronisation in between the psplash process and the bmaptool one - again sounds like overengineering to me but up to hear other opinions.

Reply:
OK, these are all non-obvious things, could you please add a comment - why the error is "passed", thanks!